src/index.js

```diff
@@ -14,7 +14,36 @@ function filter(array, callback) {
   return result;
 };
 
+function find(array, callback) {
+  for (let i = 0; i < array.length; ++i) {
+    if (callback(array[i], i, array)) return array[i];
+  }
+  return undefined;
+};
+
+function findIndex(array, callback) {
+  for (let i = 0; i < array.length; ++i) {
+    if (callback(array[i], i, array)) return i;
+  }
+  return -1;
+};
+
+function flat(array, depth) {
```

Comment:
```suggestion
function flat(array, depth = 1) {
```
The depth level specifying how deep a nested array structure should be flattened. Defaults to 1.